### ISSUE TITLE
last modified  nodeValue not persistent

### ISSUE THREAD

**Issue by nosadaniel** [cyber-geiger/toolbox-storage#47]
Describe the bug
I can't  get an accurate time of when the nodeValue was added/modified.

Environmental information
===
flutter 2.10.3
geigerApi 0.7.3
geiger_localstorage: ^0.6.42

To Reproduce
===
```
void main()async {

  late StorageController? storageController;

  group("UserService ", () {
    setUp(()async{
      GeigerApi? localApi =   await getGeigerApi("", GeigerApi.masterId, Declaration.doNotShareData);
      if(localApi != null){
        storageController = localApi.getStorage();
      }
    });

    test("add new value",()async{
      Node node = await storageController!.get(":Local");
      await node.addOrUpdateValue(NodeValueImpl("newValue", "NOsa added something" ));

      await storageController!.addOrUpdate(node);

      NodeValue? nodeValue = await storageController!.getValue(":Local", "newValue");

      int lastModified = nodeValue!.lastModified;
      String newValue = nodeValue.value;

      DateTime dateTime =
      DateTime.fromMillisecondsSinceEpoch(lastModified);

      String s = dateTime.second.toString();

      print(newValue + " $s seconds ago");



    });

    test("retrieve value added  last modified time", ()async{

      //delay by 2sec
      
      await Future.delayed(Duration(seconds: 2));
      int currentTime = DateTime.now().second;

      NodeValue? nodeValue = await storageController!.getValue(":Local", "newValue");

      int lastModified = nodeValue!.lastModified;
      String newValue = nodeValue.value;

      DateTime retrievedTime =
      DateTime.fromMillisecondsSinceEpoch(lastModified);

      String m = retrievedTime.second.toString();

      print(" retrieved $newValue $m seconds");


      //compare currentTime in seconds with  last modified time
      bool result = currentTime == retrievedTime.second ;

      expect(result,false, reason: "retrievedTimeshould be less than the current time");
    });


  });
}


```
Expected behavior
===
I should get the accurate time of when the nodeValue was added/modified.

**Comment by martingwerder** [comment]
I was able to reproduce the issue. A simplified version of your test is now in the user tests included.  The issue should be fixed in 0.6.43. Can you confirm?

**Comment by nosadaniel** [comment]
I can confirm it works.
Thanks Martin for the quick fix.
